src/events/event_processor.erl

```diff
@@ -0,0 +1,80 @@
+-module(event_processor).
+-author("Daniel Jodłoś").
+
+-export([
+  process/2,
+  async_process/2,
+  post/1
+]).
+
+-type ref() :: binary().
+-type event() :: {child | parent, removed | updated, ref(), ref(), any()}. % {node_type, operation_type, target, argument, options}
+
+-spec graph_operation(removed | updated) -> fun((ref(), ref(), any()) -> boolean()).
+graph_operation(removed) ->
+  fun (A, B, _Args) -> 
+   case graph:effective_edge_exists(A, B) of
+      {ok, true} ->
+          graph:remove_effective_edge(A, B),
+          true;
+      _Else -> false
+   end
+  end;
+
+graph_operation(updated) -> 
+  fun (A,B, Permissions) ->
+    case graph:get_effective_edge(A, B) of
+      {ok, Permissions} -> false;
+      _Else ->
+        ok = graph:update_effective_edge(A, B, Permissions),
+        true
+    end
+  end.
+
+-spec async_process(binary(), map()) -> pid().
+async_process(Vertex, Event) ->
+    spawn(?MODULE, process, [Vertex, Event]).
+
+-spec process(binary(), map()) -> ok | {error, any()}.
+process(Vertex, Event) ->
+    instrumentation:event_started(Event),
+    #{<<"type">> := Type, <<"trace">> := Trace, <<"sender">> := Sender,
+        <<"originalSender">> := OriginalSender, <<"effectiveVertices">> := EffectiveVertices} = Event,
+    EventTuple = {Type, Trace, Sender, OriginalSender, EffectiveVertices},
```

Comment:
This was just a quick fix for lint.
However, the values in this tuple DON'T actually compliant with type `event()`

So, it has to be resolved, one way or another